packages/uploadthing/src/internal/types.ts

```diff
@@ -127,5 +128,30 @@ export type inferEndpointInput<TUploader extends Uploader<any>> =
 export type inferErrorShape<TRouter extends FileRouter> =
   TRouter[keyof TRouter]["_def"]["_errorShape"];
 
-export const VALID_ACTION_TYPES = ["upload", "failure"] as const;
+export const VALID_ACTION_TYPES = [
+  "upload",
+  "failure",
+  "multipart-complete",
+] as const;
 export type ActionType = (typeof VALID_ACTION_TYPES)[number];
+
+export type UTEvents = {
```

Comment:
in the future maybe we should have a full api spec of some sort where to input to all UT api endpoints are specified and create a fetch wrapper that's typed accordingly, and automatically appends headers

Reply:
Yeah, that would be a good idea.